strawberry/field.py

```diff
@@ -134,8 +135,22 @@ def __init__(
         self.is_subscription = is_subscription
 
         self.permission_classes: List[Type[BasePermission]] = list(permission_classes)
-        self.directives = directives
-
+        self.directives = list(directives)
+        self.extensions: Optional[List[FieldExtension]] = extensions
```

Comment:
Nitpick. By doing this you don't need to do `if not self.extensions` bellow

```suggestion
        self.extensions: List[FieldExtension] = extensions or []
```